clients/gen/go.sh

```diff
@@ -25,7 +25,7 @@ readonly CLEANUP_DIRS
 # shellcheck source=./clients/gen/common.sh
 source "${CLIENTS_GEN_DIR}/common.sh"
 
-VERSION=1.1.0
+VERSION=2.0.0
```

Comment:
Shouldn't it be `2.1.0`?

Reply:
Hm. I remember we discussed versioning, but can't find it quickly. My memory is not the best recently (got some disruptions and distractions) but if I remember correctly, we agreed to only follow MAJOR release, assuming that most recent MAJOR client will work with Any MAJOR version of airflow (and in this case 2.0.0 is perfectly fine).

But I might be wrong - does anyone remember where we had that discussion and did we capture the result of it.?

Reply:
@kaxil @ashb ?

Reply:
Nop, we decided to use Major.Minor release numbers from the Airflow and only last digit denoting fixes done on the client.

Reply:
Ah. So my memory is failing me. Do you remember where the discussion was @msumit ? I think we need to capture it somewhere (unless it's already captured and I just did not find it :) ) ?

Reply:
@potiuk it was on the dev mailing list ([here](https://lists.apache.org/thread.html/r8b7f615f56eaa6d5c1955173295d70b47e038b92b84310a82e2765a9%40%3Cdev.airflow.apache.org%3E)).. I don't think that we've added it somewhere in the docs yet..

Reply:
Thanks! Yeah. I remembered everything except the final conclusion :).

I created https://github.com/apache/airflow/pull/16422 with an attempt to capture all the SemVer rules we discussed so far. Happy to iterate and merge :)